dbt_server/views.py

```diff
@@ -315,3 +315,80 @@ def get_task_status(
 ):
     task = crud.get_task(db, task_id)
     return JSONResponse(status_code=200, content={"status": task.state})
+
+
+class PostInvocationRequest(BaseModel):
+    # List of dbt command that will be sent to dbt worker for execution.
+    # E.g. ["--log-format", "json", "run", "--profiles_dir", "testdir"]
+    command: List[str]
+    # Dbt project path, if set --project-dir args will be appended into command
+    # list. If not set, dbt server will fallback to environment variable and
+    # append it. If environment variable is empty, request will be rejected.
+    # Notice user can specify --project-dir args in command directly, in that
+    # case, we will respect user request and won't append anything.
+    project_path: Optional[str]
+    # Optional, if set dbt worker will trigger callback with task id and task
+    # status when task status is updated.
+    callback_url: Optional[str]
+
+
+class PostInvocationResponse(BaseModel):
+    # Unique task id of invocation.
+    task_id: str
+    # Absolute local path pointed to dbt.log file for the corresponding task.
+    # It's available if user doesn't pass in --log-path args in command.
+    # Notice it's not guaranteed dbt.log will always be generated by core(e.g.
+    # critical error occurs).
+    log_path: Optional[str]
+
+
+@app.post("/invocation")
```

Comment:
Are you planning to eventually rename this to the existing `/async/dbt` endpoint? I do think it makes sense to start over on these endpoints rather than rewrite the existing ones for now, just trying to get a sense of how you envision this looking at its end state. For example if all invocations are now going to go through this endpoint, we'll still need `state_id` and `task_id` as optional params on the `PostInvocationRequest` model for the runtime gateway usecase.

Reply:
I know now we're going to be using the celery `task_id`, so we'll have to think about how that's going to work with the runtime-gateway -- we previously planned to have the RG generate an id and send it as part of the request so that we could maintain a source of truth in the runtime database

Reply:
Offline chatted with Rachel, in short time we won't touch existing endpoints, we will revisit them later.

RG generated id seems to be an metadata of the task, where we don't store any metadata at this moment, but I foresee we do need such metadata in future. In short, we are okay to maintain a
- a mapping from celery task id to some metadata(RG task id + log path, I'm struggled with log path for a while as you can see it's not returned in GET endpoint because we don't save that in POST and user can overwrite it)
- a mapping between RG task id to celery task id, that's trivial. 
- inside dbt server, celery task id is always the primary key.